app/services/exporters/generate_users_participations_csv.rb

```diff
@@ -2,38 +2,39 @@ module Exporters
   class GenerateUsersParticipationsCsv < GenerateUsersCsv
     private
 
-    def each_element(&block)
-      @users.each do |user|
-        user.participations
-            .joins(:rdv)
-            .order("rdvs.starts_at desc")
-            .where(rdvs: { organisation_id: @agent.organisation_ids })
-            .to_a.each(&block)
+    def each_element(&)
+      filtered_participations.each(&)
+    end
+
+    def filtered_participations
+      if @motif_category
+        @participations.where(user_id: @user_ids)
+                       .joins(rdv: { motif: :motif_category })
+                       .where(motif_categories: { id: @motif_category.id })
+      else
+        @participations.where(user_id: @user_ids)
+                       .joins(:rdv)
       end
+        .where(rdvs: { organisation_id: @agent.organisation_ids })
+        .order("rdvs.starts_at desc")
```

Comment:
nitpick: la lisibilité de la requête est un peu venère comme ça, il vaudrait peut-être mieux utiliser des variables intermédiaires.
```suggestion
      participations =  
      if @motif_category
        @participations.where(user_id: @user_ids)
                       .joins(rdv: { motif: :motif_category })
                       .where(motif_categories: { id: @motif_category.id })
      else
        @participations.where(user_id: @user_ids).joins(:rdv)
      end
       
      participations.where(rdvs: { organisation_id: @agent.organisation_ids }).order("rdvs.starts_at desc")
```

Reply:
Mouais... Si ca peut te faire plaisir 🤷‍♂️